sql/core/src/main/scala/org/apache/spark/sql/catalog/Catalog.scala

```diff
@@ -589,4 +589,25 @@ abstract class Catalog {
    * @since 2.0.0
    */
   def refreshByPath(path: String): Unit
+
+  /**
+   * Returns the current default catalog in this session.
+   *
+   * @since 3.4.0
+   */
+  def currentCatalog(): String
```

Comment:
Would need to 1. have a default implementation for backward compatibility, or 2. exclude this from binary compatibility check.

The real error is:

```
[error] spark-sql: Failed binary compatibility check against org.apache.spark:spark-sql_2.12:3.2.0! Found 3 potential problems (filtered 602)
[error]  * abstract method currentCatalog()java.lang.String in class org.apache.spark.sql.catalog.Catalog is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("org.apache.spark.sql.catalog.Catalog.currentCatalog")
[error]  * abstract method setCurrentCatalog(java.lang.String)Unit in class org.apache.spark.sql.catalog.Catalog is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("org.apache.spark.sql.catalog.Catalog.setCurrentCatalog")
[error]  * abstract method listCatalogs()org.apache.spark.sql.Dataset in class org.apache.spark.sql.catalog.Catalog is present only in current version
[error]    filter with: ProblemFilters.exclude[ReversedMissingMethodProblem]("org.apache.spark.sql.catalog.Catalog.listCatalogs")
```

Reply:
Do we need backward compatibility here? This is an interface that Spark implements and users call. Adding a new method shouldn't break anything.

Reply:
Yeah, acutlaly not sure why it complains.

Reply:
anyway, let's update the build file to make mima happy